java/code/src/com/redhat/rhn/manager/formula/FormulaManager.java

```diff
@@ -348,6 +348,15 @@ public void enableFormula(String minionId, String formulaName) throws IOExceptio
                 .collect(Collectors.toList());
     }
 
+    public List<Map<String, FormulaData>> getCombinedFormulaDataForSystems(User user, List<Integer> systemIDs) {
+        return systemIDs.stream()
+                .map(Integer::longValue)
+                .map(sID -> FormulaFactory.getCombinedFormulasByServerId(sID).stream()
+                .collect(Collectors.toMap(formula -> formula,
+                        formula -> getCombinedFormulaDataForSystems(user, List.of(sID), formula).get(0))))
```

Comment:
Is this call expected to perform well on large setups?

Reply:
I haven't measured performance but taking into account that all the information is available locally on the disk it should not be a problem.